api/src/main/java/org/openmrs/module/systemmonitor/api/db/hibernate/HibernateSystemMonitorDAO.java

```diff
@@ -119,6 +119,26 @@ public String getToday() {
 	}
 
 	@Override
+	public String getYesterdayStartDate() {
+		Calendar calendar = Calendar.getInstance(Context.getLocale());
+
+		calendar.add(Calendar.DATE, -1);
+		resetDateTimes(calendar);
+
+		return sdf.format(calendar.getTime());
+	}
+
+	@Override
+	public String getYesterdayEndDate() {
+		Calendar calendar = Calendar.getInstance(Context.getLocale());
+
+		resetDateTimes(calendar);
+		calendar.add(Calendar.MILLISECOND, -1);
```

Comment:
I had actually not reviewed this, am thinking that we probably need to first of all subtract one from DAY_OF_MONTH to fall onto yesterday. am seemingly agreeing that after going backward by day, we can then just subtract one from milli_secs, i will actually first need to check this out, you could probably test this whole method in a unit test